ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/metrics/DeltaFilesMetricReporter.java

```diff
@@ -398,13 +156,6 @@ private static long getBaseSize(AcidDirectory dir) throws IOException {
     return baseSize;
   }
 
-  private static long getModificationTime(AcidUtils.ParsedDirectory dir, FileSystem fs) throws IOException {
-    return dir.getFiles(fs, Ref.from(false)).stream()
-      .map(HadoopShims.HdfsFileStatusWithId::getFileStatus)
-      .mapToLong(FileStatus::getModificationTime)
-      .max()
-      .orElse(new Date().getTime());
-  }
 
   private static long getDirSize(AcidUtils.ParsedDirectory dir, FileSystem fs) throws IOException {
```

Comment:
I'm a bit concerned about this method. Do you think it might slow down the Initiator/Worker/Cleaner?

Reply:
It will slow it down, but there is no other way we could calculate the directory size.

Reply:
Do you think that collecting metrics about small deltas is worth the amount this might slow down the Initiator/Worker/Cleaner?

Reply:
I think we should collect the small delta metrics because they can indicate issues with streaming (I wish I had such a metric when I investigated the Northfolk escalation :) )
Also, we are doing the same calculation in the `Initiator` code (`Initiator#sumDirSize`) to determine the compaction type.